puppet/spec/unit/indirector/catalog/puppetdb_spec.rb

```diff
@@ -702,5 +702,55 @@ def test_file_require(resource_title, require_title)
           "catalog_uuid", "producer"]
       end
     end
+
+    describe "#redact_sensitive_params" do
+      let(:secret) {'xyzzy XKPGMKIB253ZVJHKOKZZXPQSSE'}
+      it 'should remove any sensitive resource parameters' do
+        input = {'tags'=>Puppet::Util::TagSet.new(['settings']),
+                 'name'=>'my_agent',
+                 'version'=>1490991352,
+                 'code_id'=>nil,
+                 'catalog_uuid'=>'aa4759d3-f1f1-47a0-925c-a4acd0c1b4ed',
+                 'catalog_format'=>1,
+                 'environment'=>'production',
+                 'resources'=>
+                 [{'type'=>'Stage',
+                   'title'=>:main,
+                   'tags'=>Puppet::Util::TagSet.new(['stage']),
+                   'exported'=>false,
+                   'parameters'=>{:name=>'main'}},
+                  {'type'=>'Class',
+                   'title'=>'Settings',
+                   'tags'=>Puppet::Util::TagSet.new(['class', 'settings']),
+                   'exported'=>false},
+                  {'type'=>'Class',
+                   'title'=>:main,
+                   'tags'=>Puppet::Util::TagSet.new(['class']),
+                   'exported'=>false,
+                   'parameters'=>{:name=>'main'}},
+                  {'type'=>'Notify',
+                   'title'=>'hi',
+                   'tags'=>Puppet::Util::TagSet.new(['notify', 'hi', 'class']),
+                   'file'=> 'site.pp',
+                   'line'=>1,
+                   'exported'=>false,
+                   'parameters'=>{:message=>secret},
+                   'sensitive_parameters'=>[:message]}],
+                 'edges'=>
+                 [{'source'=>'Stage[main]', 'target'=>'Class[Settings]'},
+                  {'source'=>'Stage[main]', 'target'=>'Class[main]'},
+                  {'source'=>'Class[main]', 'target'=>'Notify[hi]'}],
+                 'classes'=>['settings']}
+        def has_secret?(r)
```

Comment:
This surprised me here, do we have this in other tests? My guess is it'd be more typical to have this fn separate (like a defn)